Lib/test/test_future_stmt/test_future.py

```diff
@@ -18,21 +20,43 @@ def get_error_location(msg):
 
 class FutureTest(unittest.TestCase):
 
-    def check_syntax_error(self, err, basename, lineno, offset=1):
-        self.assertIn('%s.py, line %d' % (basename, lineno), str(err))
-        self.assertEqual(os.path.basename(err.filename), basename + '.py')
+    def check_syntax_error(self, err, basename, lineno,
+                           message=TOP_LEVEL_MSG, offset=1):
+        if basename != '<string>':
+            basename += '.py'
+
+        self.assertEqual(f'{message} ({basename}, line {lineno})', str(err))
+        self.assertEqual(os.path.basename(err.filename), basename)
         self.assertEqual(err.lineno, lineno)
         self.assertEqual(err.offset, offset)
 
-    def test_future1(self):
-        with import_helper.CleanImport('test.test_future_stmt.future_test1'):
-            from test.test_future_stmt import future_test1
-            self.assertEqual(future_test1.result, 6)
+    def assertSyntaxError(self, code, lineno,
+                          message=TOP_LEVEL_MSG, offset=1,
+                          *,
+                          parametrize_docstring=True):
+        code = dedent(code)
+        for trim_docstring in ([False, True] if parametrize_docstring else [False]):
+            with self.subTest(code=code, trim_docstring=trim_docstring):
+                if trim_docstring:
+                    code = os.linesep.join(code.splitlines()[2:])
+                    lineno -= 2
+                with self.assertRaises(SyntaxError) as cm:
+                    exec(code)
+                self.check_syntax_error(cm.exception, "<string>", lineno,
+                                        message, offset=offset)
+
+    def test_import_nested_scope_twice(self):
+        # Import the name nested_scopes twice to trigger SF bug #407394
+        with import_helper.CleanImport(
+            'test.test_future_stmt.import_nested_scope_twice',
+        ):
+            from test.test_future_stmt import import_nested_scope_twice
+        self.assertEqual(import_nested_scope_twice.result, 6)
 
-    def test_future2(self):
-        with import_helper.CleanImport('test.test_future_stmt.future_test2'):
-            from test.test_future_stmt import future_test2
-            self.assertEqual(future_test2.result, 6)
+    def test_nested_scope(self):
+        with import_helper.CleanImport('test.test_future_stmt.nested_scope'):
+            from test.test_future_stmt import nested_scope
+        self.assertEqual(nested_scope.result, 6)
```

Comment:
6 looks like magic. would you mind to write ``lineno=6`` instead? Maybe mark this parameter as keyword-only. You may mark all parameters as keyword only, except of the first one, *code*.

Reply:
But, it is just 6 :)
This is `assertEqual`. I will do it for other places, sure.

Reply:
Sorry, I added my comment at the wrong places! You addressed my remark anyway, thanks.